### PR TITLE
[Java] Some improvements to `parse_decimal` function and bindings for `is_fixed_point`

### PR DESCRIPTION
**Pull request by razajafri**
This PR adds Java bindings for `is_fixed_point`

### CODE REVIEW

**Comment by razajafri** [comment]
@revans2 most of what we talked about was implemented already. PTAL and let me know if I missed anything

File: cpp/include/cudf/strings/detail/convert/fixed_point.cuh
Comment (revans2) [inline cpp/include/cudf/strings/detail/convert/fixed_point.cuh:55]: Why do we need this change? The data is incorrect. Checking the format of the data should be handled by `is_fixed_point` not here.

Comment (davidwendt) [reply cpp/include/cudf/strings/detail/convert/fixed_point.cuh:55]: Bobby is right. This if-check should not go here.

**Review by davidwendt — CHANGES_REQUESTED** [review]

File: cpp/include/cudf/strings/detail/convert/fixed_point.cuh
Comment (davidwendt) [inline cpp/include/cudf/strings/detail/convert/fixed_point.cuh:139]: This if-statement is also not be necessary. The current code logic handles empty strings already.

**Review by ttnghia — APPROVED** [review]

**Comment by razajafri** [comment]
@gpucibot merge

**Comment by ttnghia** [comment]
This PR is no longer a libcudf PR and just requires java codeowner approval.

File: java/src/main/java/ai/rapids/cudf/ColumnView.java
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:352]: nit: can we clarify this a bit more? It is not clear what the input type should be or really what this does in terms of checks. The following from the C++ docs I think explains it better, but could possibly be paired down a bit too. Also it looks like it still is explaining the old behavior. Wasn't it updated to work with exponent format? like '+17E30'?

```
 * @brief Returns a boolean column identifying strings in which all
 * characters are valid for conversion to fixed-point.
 *
 * The output row entry is set to `true` if the corresponding string element
 * has at least one character in [+-0123456789.]. The optional sign character
 * must only be in the first position. The decimal point may only appear once.
 * Also, the integer component must fit within the size limits of the
 * underlying fixed-point storage type. The value of the integer component
 * is based on the scale of the `decimal_type` provided.
 *
 * @code{.pseudo}
 * Example:
 * s = ['123', '-456', '', '1.2.3', '+17E30', '12.34' '.789', '-0.005]
 * b = is_fixed_point(s)
 * b is [true, true, false, false, false, true, true, true]
 * @endcode
 *
 * Any null entries result in corresponding null entries in the output column.
```

**Review by revans2 — APPROVED** [review]
Just a documentation nit

**Review by davidwendt — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

File: java/src/main/java/ai/rapids/cudf/ColumnView.java
Comment (razajafri) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java:352]: I have updated both the docs. PTAL before I merge

**Comment by razajafri** [comment]
rerun tests

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by razajafri** [comment]
rerun tests

**Comment by razajafri** [comment]
@gpucibot merge

**Pull request merged**
